src/transcript_tool.py

```diff
@@ -56,39 +88,153 @@ def main(in_path: str, out_path: str, model_path: str) -> None:
+    programme = AudioSegment.from_mp3(audio_file)
+
+    for i in range(0, len(programme), minute):
+        slice_mp3 = programme[i : i + minute]
+        start_second = i / 1000
+        slice_name = os.path.join(
+            audio_file_slice_folder, base_name + f"_{start_second:07}.mp3"
+        )
+        slice_mp3.export(slice_name, format="mp3")
+        logging.info("Just saved slice: %s", slice_name)
+
+
+def segmenting(slices_folder: str) -> None:
+    "Segment the slices further, while filering out noise and music"
+    seg = Segmenter()
+    slice_list = [
+        os.path.join(slices_folder, f) for f in sorted(os.listdir(slices_folder))
+    ]
+    slice_list = [f for f in slice_list if os.path.isfile(f)]
+
+    _, folder_name = separate_fileneme(slices_folder)
+    audio_file_segment_folder = os.path.join(
+        os.path.abspath("../data/interim/segments"), folder_name
+    )
+    os.makedirs(audio_file_segment_folder, exist_ok=True)
+
+    for slice_mp3 in slice_list:
+        segmentation = seg(slice_mp3)
+        programme = AudioSegment.from_mp3(slice_mp3)
+        _, outfile = separate_fileneme(slice_mp3)
+        # avoid fails on '_' in filenames
+        outfile_parts = outfile.split("_")
+        slice_base = "_".join(outfile_parts[:-1])
+        slice_start_sec = outfile_parts[-1]
+
+        for segment in segmentation:
+            if segment[0] == "male" or segment[0] == "female":
+                start, end = timestamp_segment(
+                    float(slice_start_sec), segment[1], segment[2]
+                )
+                outfile = f"{slice_base}_{start}-{end}.mp3"
+                outfile = os.path.join(audio_file_segment_folder, outfile)
+
+                onset = segment[1] * 1000
+                offset = segment[2] * 1000
+                speech = programme[onset:offset]
+                speech.export(outfile, format="mp3")
+                logging.info("Saved segment: %s", outfile)
+
+
+def separate_fileneme(verbose_filename: str) -> tuple[str, str]:
+    """Separate the path to a file and the filename without extension.
+    return (path_to_file, filename_without_extension)"""
+    return os.path.split(os.path.splitext(verbose_filename)[0])
+
+
+def timestamp_segment(offset: float, seg_start: int, seg_end: int) -> tuple[str, str]:
+    "Create timestamp from offset and time inteval start/end times. All inputs should be in seconds."
+    start_stamp = str(datetime.timedelta(seconds=offset + seg_start)).split(
+        ".", maxsplit=1
+    )[0]
+    end_stamp = str(datetime.timedelta(seconds=offset + seg_end)).split(
+        ".", maxsplit=1
+    )[0]
+    return (start_stamp, end_stamp)
+
+
+def is_processed(mp3: str, out_path: str) -> bool:
+    "Check wether the given audio file already has a transcript."
+    _, base_name = separate_fileneme(mp3)
+    file_name = os.path.join(out_path, base_name + ".txt")
+    return os.path.isfile(file_name)
 
 
 if __name__ == "__main__":
```

Comment:
maybe we can refactor the functions into an other module and here we just import it and provide the CLI in main

Reply:
separated utility function to utils.py as a first step ([eba5a6d](https://github.com/Koffair/radioship_transcripter/pull/2/commits/eba5a6d6b8af4b60b5d15a69dc34dc268e1530d9))
(had some disagreement with mypy on the proper way to import, but figured out)